### PR TITLE
Feature/ADF-421/Extend remote list with extra optional parameter to define dependency field

### PR DESCRIPTION
**Pull request by shpran**
**Relates to:** _[ADF-421](https://oat-sa.atlassian.net/browse/ADF-421)_

- [x] Extend remote list with extra optional parameter to define dependency field
- [x] Add and configure feature flag for this feature

---

**Require PR:** https://github.com/oat-sa/tao-core/pull/3045

### CODE REVIEW

File: controller/Lists.php [hunk 1]
```diff
@@ -186,22 +191,30 @@ public function reloadRemoteList(ValueCollectionService $valueCollectionService,
         $this->returnJson(['saved' => $saved]);
     }
 
-    private function createList(string $label, string $source, string $labelPath, string $uriPath): RdfClass
+    private function createList(array $values): RdfClass
```
Comment (boajer) [inline controller/Lists.php:194]: Can't those method be called with empty or irrelevant array passed as it parameter?

**Review by boajer — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: controller/Lists.php [hunk 2]
```diff
@@ -186,22 +191,30 @@ public function reloadRemoteList(ValueCollectionService $valueCollectionService,
...
         $propertySource = new RdfProperty(RemoteSourcedListOntology::PROPERTY_ITEM_LABEL_PATH);
-        $class->setPropertyValue($propertySource, $labelPath);
+        $class->setPropertyValue($propertySource, $values[tao_actions_form_RemoteList::FIELD_ITEM_LABEL_PATH]);
 
         $propertySource = new RdfProperty(RemoteSourcedListOntology::PROPERTY_ITEM_URI_PATH);
-        $class->setPropertyValue($propertySource, $uriPath);
+        $class->setPropertyValue($propertySource, $values[tao_actions_form_RemoteList::FIELD_ITEM_URI_PATH]);
+
+        if ($this->isListsDependencyEnabled()) {
+            $propertySource = new RdfProperty(RemoteSourcedListOntology::PROPERTY_DEPENDENCY_ITEM_URI_PATH);
```
Comment (gabrielfs7) [inline controller/Lists.php:212]: This means we are duplicating data from the new table also in statements table?

**Review by gabrielfs7 — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Review by chinnu-focaloid — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: composer.json [hunk 3]
```diff
@@ -23,7 +23,7 @@
     "require": {
         "oat-sa/oatbox-extension-installer": "~1.1||dev-master",
         "oat-sa/generis" : ">=14.0.0",
-        "oat-sa/tao-core" : ">=47.0.0"
+        "oat-sa/tao-core" : ">=48.23.0"
```
Comment (gabrielfs7) [inline composer.json:26]: - [ ] Make sure to validate the version before merge

**Pull request merged**